instrumentor/controllers/workloadmigrations/controllers.go

```diff
@@ -150,11 +155,26 @@ func CreateOrUpdateSourceForObject(ctx context.Context,
 		}
 	}
 	source.Spec.DisableInstrumentation = disableInstrumentation
+	// migrate the reported name from the workload to the Source only if the Source is not a namespace Source
+	// and the workload Source does not already have a service name.
+	//
+	// If a user have set the annotation on the workload we will use it in the source only if no reported name is set.
+	// If that happens once, and the reported name was taken from the annotation,
+	// any more changes to the annotation will not be reflected in the source.
+	// This is valid, since the annotation is deprecated and we want to encourage users to use Source CR.
+	if kind != workload.WorkloadKindNamespace && source.Spec.OtelServiceName == "" {
```

Comment:
do we need to check here that `serviceName` is not empty string?
how will it act if the annotation is missing to begin with?